dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+        """Initialize parameters for learning skeleton.
+
+        Parameters
+        ----------
+        nodes : list of nodes
+            The list of nodes that will be present in the learned skeleton graph.
+        """
+        # error checks of passed in arguments
+        if self.max_combinations is not None and self.max_combinations <= 0:
+            raise RuntimeError(f"Max combinations must be at least 1, not {self.max_combinations}")
+
+        if self.skeleton_method not in ACCEPTED_MARKOVIAN_SKELETON_METHODS:
+            raise ValueError(
+                f"Skeleton method must be one of {ACCEPTED_MARKOVIAN_SKELETON_METHODS}, "
+                f"not {self.skeleton_method}."
+            )
+
+        if self.sep_set is None:
+            # keep track of separating sets
+            self.sep_set_ = defaultdict(lambda: defaultdict(list))
+        else:
+            self.sep_set_ = self.sep_set
+
+        # control of the conditioning set
+        if self.max_cond_set_size is None:
+            self.max_cond_set_size_ = np.inf
+        else:
+            self.max_cond_set_size_ = self.max_cond_set_size
+        if self.min_cond_set_size is None:
+            self.min_cond_set_size_ = 0
+        else:
+            self.min_cond_set_size_ = self.min_cond_set_size
+        if self.max_combinations is None:
+            self.max_combinations_ = np.inf
+        else:
+            self.max_combinations_ = self.max_combinations
+
+    def fit(self, context: Context):
+        """Run structure learning to learn the skeleton of the causal graph.
+
+        Parameters
+        ----------
+        context : Context
+            A context object.
+        """
+        # get the initialized graph
+        adj_graph = context.init_graph
+        nodes = adj_graph.nodes
+        X = context.data
+
+        # track progress of the algorithm for which edges to remove to ensure stability
+        self.remove_edges = set()
+
+        # initialize learning parameters
+        self._initialize_params()
+
+        # the size of the conditioning set will start off at the minimum
+        size_cond_set = self.min_cond_set_size_
+
+        # store the absolute value of test-statistic values for every single
+        # candidate parent-child edge (X -> Y)
+        nx.set_edge_attributes(adj_graph, np.inf, "test_stat_abs")
+
+        # store the actual minimum test-statistic/pvalue value for every
+        # single candidate parent-child edge
+        nx.set_edge_attributes(adj_graph, -1e-5, "pvalue")
+        nx.set_edge_attributes(adj_graph, np.inf, "test_stat_min")
+
+        # store the list of potential adjacencies for every node
+        # which is tracked and updated in the algorithm
+        adjacency_mapping: Dict[Any, List] = dict()
+        for node in nodes:
+            adjacency_mapping[node] = [
+                other_node for other_node in adj_graph.neighbors(node) if other_node != node
+            ]
+
+        logger.info(
```

Comment:
There should be an option to turn off logging information. Maybe make the logger an optional parameter that can be passed or add flag to turn it off. (This applies to all logging/print information)

Reply:
Doesn't logging allow the user to set "logging level" to say "error" and they can automatically not see any of the info logs?

Or do you mean just encompass `logger.info` messages perhaps in some type of `verbose` conditional check?

Reply:
I don't have so many experience with the logger, but since you initiate a new logger at the beginning, I am not sure if this would honor your settings (or if the settings are global). Maybe worth to simply try it out :)